lib/queue.js

```diff
@@ -2,11 +2,16 @@
 
 const queue = function() {
   this.tasks = [];
+  this.onDone = result => result;
 };
 
-queue.prototype.pushTasks = function(task) {
-  if (typeof task === 'function') this.tasks.push(task);
-  else task.forEach(task => this.pushTasks(task));
+queue.prototype.pushTask = function(fn, args = []) {
+  const task = fn.bind(this, ...args);
```

Comment:
Wouldn't it be better to use partial application here instead of bind?
```javascript
const partial = (fn, ...args) => (...rest) => fn(...args, ...rest);
```
```suggestion
  const task = partial(fn, ...args);
```